examples/oink-oink/src/logic.ts

```diff
@@ -0,0 +1,169 @@
+        const actor = game.players.find((player) => player.actor)
+
+        if (!player || !actor) throw Rune.invalidAction()
+
+        player.score += 1
+        player.latestTurnScore += 1
+        player.latestRoundScore += 1
+        actor.score += 1
+        actor.latestTurnScore += 1
+        actor.latestRoundScore += 1
+
+        game.currentTurn.animal = getRandomItem(animals)
+        game.currentTurn.emotion = getRandomItem(emotions)
+
+        if (!game.currentTurn.timerStartedAt) throw Rune.invalidAction()
+        game.currentTurn.timerStartedAt += displayCorrectGuessFor
+      }
+    },
+    nextRound: (_, { game }) => {
+      if (game.round + 1 === numRounds) throw Rune.invalidAction()
+      if (game.currentTurn?.stage !== "result") throw Rune.invalidAction()
+
+      for (const player of game.players) {
+        player.latestRoundScore = 0
+      }
+
+      game.round += 1
+      setActor(game, "first")
+      newTurn(game)
+    },
+  },
+  events: {
+    playerLeft: (playerId, { game }) => {
+      game.players = game.players.filter((player) => player.id !== playerId)
+
+      startGameCheck(game)
+
+      if (
+        game.currentTurn &&
+        (game.currentTurn.stage === "acting" ||
+          game.currentTurn.stage === "countdown")
+      ) {
+        if (isLastActor(game)) {
+          game.currentTurn.stage = "result"
+        } else {
+          setActor(game, "next")
+          newTurn(game)
+        }
+      }
+    },
+  },
+  update: ({ game }) => {
+    if (!game.currentTurn) return
+
+    if (
+      game.currentTurn.stage === "countdown" &&
+      game.currentTurn.timerStartedAt &&
+      Rune.gameTimeInSeconds() >=
+        game.currentTurn.timerStartedAt + turnCountdown
+    ) {
+      game.currentTurn.stage = "acting"
+      game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
+    }
+
+    if (
+      game.currentTurn.stage === "acting" &&
+      game.currentTurn.timerStartedAt &&
+      Rune.gameTimeInSeconds() >= game.currentTurn.timerStartedAt + turnDuration
+    ) {
+      game.currentTurn.stage = "endOfTurn"
+      game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
+    }
+
+    if (
+      game.currentTurn.stage === "endOfTurn" &&
+      game.currentTurn.timerStartedAt &&
+      Rune.gameTimeInSeconds() >=
+        game.currentTurn.timerStartedAt + endOfTurnDuration
+    ) {
+      for (const player of game.players) {
+        player.latestTurnScore = 0
+      }
+
+      if (isLastActor(game)) {
+        game.currentTurn.stage = "result"
+
+        if (game.round + 1 === numRounds) {
+          game.gameOver = true
+          Rune.gameOver({
+            players: game.players.reduce(
+              (acc, player) => ({
+                ...acc,
+                [player.id]: player.score,
+              }),
+              {}
+            ),
+            delayPopUp: true,
+          })
+        }
+      } else {
+        setActor(game, "next")
+        newTurn(game)
+      }
+    }
+  },
```

Comment:
```suggestion
  update: ({ game }) => {
    if (!game.currentTurn || !game.currentTurn.timerStartedAt) return

    switch (game.currentTurn.stage) {
      case "countdown":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + turnCountdown
        ) {
          game.currentTurn.stage = "acting"
          game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
        }
        return
      case "acting":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + turnDuration
        ) {
          game.currentTurn.stage = "endOfTurn"
          game.currentTurn.timerStartedAt = Rune.gameTimeInSeconds()
        }
        return
      case "endOfTurn":
        if (
          Rune.gameTimeInSeconds() >=
          game.currentTurn.timerStartedAt + endOfTurnDuration
        ) {
          for (const player of game.players) {
            player.latestTurnScore = 0
          }

          if (isLastActor(game)) {
            game.currentTurn.stage = "result"

            if (game.round + 1 === numRounds) {
              game.gameOver = true
              Rune.gameOver({
                players: game.players.reduce(
                  (acc, player) => ({
                    ...acc,
                    [player.id]: player.score,
                  }),
                  {}
                ),
                delayPopUp: true,
              })
            }
          } else {
            setActor(game, "next")
            newTurn(game)
          }
        }
        return
    }
  },
```

IMO we can refactor this to make it clearer by:
- moving up `!game.currentTurn.timerStartedAt` into an early return
- converting the if conditions to a switch case

I think this helps make it clear that the logic in the stages are mutually exclusive and see what the logic is for each stage to move to the next. But it's a matter of personal preference 😄

Reply:
Good idea, thanks! Done